src/libs/algolia/__tests__/fetchAlgolia.test.ts

```diff
@@ -416,6 +417,154 @@ describe('fetchOffer', () => {
+
+      fetchOffer({
+        parameters: { ...baseParams, query, offerGenreTypes } as SearchParametersQuery,
+        userLocation: null,
+        isUserUnderage: false,
+      })
+
+      expect(search).toHaveBeenCalledWith(query, {
+        facetFilters: [['offer.isEducational:false'], ['offer.movieGenres:DRAMA']],
+        numericFilters: [['offer.prices: 0 TO 300']],
+        page: 0,
+        attributesToHighlight: [],
+        attributesToRetrieve: offerAttributesToRetrieve,
+        clickAnalytics: true,
+      })
+    })
+
+    it('should fetch with facetFilters parameter when one bookMacroSection is provided', () => {
+      const query = 'searched query'
+      const offerGenreTypes = [{ key: GenreType.BOOK, value: 'Droit', name: 'Droit' }]
+
+      fetchOffer({
+        parameters: { ...baseParams, query, offerGenreTypes } as SearchParametersQuery,
+        userLocation: null,
+        isUserUnderage: false,
+      })
+
+      expect(search).toHaveBeenCalledWith(query, {
+        facetFilters: [['offer.isEducational:false'], ['offer.bookMacroSection:Droit']],
+        numericFilters: [['offer.prices: 0 TO 300']],
+        page: 0,
+        attributesToHighlight: [],
+        attributesToRetrieve: offerAttributesToRetrieve,
+        clickAnalytics: true,
+      })
+    })
+
+    it('should fetch with facetFilters parameter when one showType is provided', () => {
+      const query = 'searched query'
+      const offerGenreTypes = [
+        { key: GenreType.SHOW, value: 'Arts de la rue', name: 'Arts de la rue' },
+      ]
+
+      fetchOffer({
+        parameters: { ...baseParams, query, offerGenreTypes } as SearchParametersQuery,
+        userLocation: null,
+        isUserUnderage: false,
+      })
+
+      expect(search).toHaveBeenCalledWith(query, {
+        facetFilters: [['offer.isEducational:false'], ['offer.showType:Arts de la rue']],
+        numericFilters: [['offer.prices: 0 TO 300']],
+        page: 0,
+        attributesToHighlight: [],
+        attributesToRetrieve: offerAttributesToRetrieve,
+        clickAnalytics: true,
+      })
+    })
+
+    it('should fetch with facetFilters parameter when one musicType is provided', () => {
+      const query = 'searched query'
+      const offerGenreTypes = [{ key: GenreType.MUSIC, value: 'Pop', name: 'Pop' }]
+
+      fetchOffer({
+        parameters: { ...baseParams, query, offerGenreTypes } as SearchParametersQuery,
+        userLocation: null,
+        isUserUnderage: false,
+      })
+
+      expect(search).toHaveBeenCalledWith(query, {
+        facetFilters: [['offer.isEducational:false'], ['offer.musicType:Pop']],
+        numericFilters: [['offer.prices: 0 TO 300']],
+        page: 0,
+        attributesToHighlight: [],
+        attributesToRetrieve: offerAttributesToRetrieve,
+        clickAnalytics: true,
+      })
+    })
+
+    it('should fetch with facetFilters parameter when unknown genre type is provided', () => {
```

Comment:
D'une manière générale est-ce qu'il ne serait pas prudent de s'assurer du test "inverse", par exemple 
should fetch **without** facetFilters parameter when **no** musicType is provided

?

Reply:
Aucun des tests de ce fichier n'ont été fait comme ça, j'ai l'impression que de manière générale on teste moins comme ça mais est-ce que tu veux que je les rajoute ?

Reply:
Pas forcément, je dirai que c'est un sujet à discuter.